legion/engine/physics/data/colliders/collider_data.hpp

```diff
@@ -40,11 +40,16 @@ namespace legion::physics
 
         ConvexColliderData(const math::vec3& offset,const math::quat& rotation, const math::vec3& boxExtents) noexcept;
 
+        ConvexColliderData(const math::vec3& offset, const math::quat& rotation, void* internalConvexColliderStructure);
```

Comment:
the void* is to store the 'backend' convex collider implementation, so that this class can store a 'physx convex collider' or a 'diviner convex collider' or a 'bullet convex collider' etc

Reply:
iirc the general practice for this is either `using ConvexCollider = void;` `const ConvexCollider* ccPtr` or `using ConvexColliderPtr = void*` `const ConvexColliderPtr ccPtr`

Reply:
also arguably, this should be a struct?

Reply:
> also arguably, this should be a struct?
as of right now yes, but future subfeatures will add more logic to it